### ISSUE TITLE
Video Playback Support on Linux

### ISSUE THREAD

**Issue by Frooxius** [Neos-Metaverse/NeosPublic#2762]
### Is your feature request related to a problem? Please describe.

Currently video playback isn't supported in the native Linux builds, unlike the Windows or Android ones.

### Describe the solution you'd like

Integrate the libVLC library and ensure it works on the Linux build. Currently this is blocked by the library itself not supporting Linux yet, but it is on their roadmap: https://code.videolan.org/videolan/vlc-unity

Since it's an open source project, efforts and contributions would be welcome as they would help get this issue unblocked sooner!

We can also explore enabling the native Unity video playback, but at the moment I'm not sure how well is that integrated on Linux. The documentation doesn't even seem to mention the supported formats for it, so it looks like the situation is quite poor.

### Describe alternatives you've considered

We could also explore other video playback libraries that already support Linux, but ideally we'd build everything around the same one to ensure consistency and feature parity between platforms.

### Additional context

Splitting this up from the megathread: https://github.com/Neos-Metaverse/NeosPublic/issues/149

**Comment by mrxz** [comment]
Adding Linux support to vlc-unity is pretty straightforward. An (outdated) merge request adding Android support to the plugin ([here](https://code.videolan.org/videolan/vlc-unity/-/merge_requests/15)) can be used as reference since that also uses OpenGL. It's just a matter of replacing EGL with something like GLX and updating the Makefile.

When I tried that for a hobby project, I did need to make two additional tweaks/hacks to LibVLCSharp and libvlccore. One related to needing a library path under Windows but explicitly _not_ expecting one under Linux. The other to workaround an issue with the tcmalloc version bundled in Unity. I never bothered cleaning that up and never released the code as a result as it doesn't work without these tweaks.

Once #660 is done, it will be interesting to see if the above can be made to work with Neos. Having an actual application work with it, will add a bit of validity to the changes and make it worthwhile to clean up and contribute back. Not to mention that it would be nice to have working video playback in a social VR application under Linux.

**Comment by Frozenreflex** [comment]
A fix for this is included in my [NeosVideoPlayerFix](https://github.com/Frozenreflex/NeosVideoPlayerFix) mod.
